pkg/command/service/measure_test.go

```diff
@@ -0,0 +1,171 @@
+	fakenetworkingv1alpha1 "knative.dev/networking/pkg/client/clientset/versioned/typed/networking/v1alpha1/fake"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+	servingv1fake "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1/fake"
+)
+
+func TestNewServiceMeasureCommand(t *testing.T) {
+	t.Run("incompleted or wrong args for service measure", func(t *testing.T) {
+		client := k8sfake.NewSimpleClientset()
+
+		p := &pkg.PerfParams{
+			ClientSet: client,
+		}
+
+		cmd := NewServiceMeasureCommand(p)
+
+		_, err := testutil.ExecuteCommand(cmd)
+		assert.ErrorContains(t, err, "'service measure' requires flag(s)")
+
+		_, err = testutil.ExecuteCommand(cmd, "--range", "1200", "--namespace", "ns")
+		assert.ErrorContains(t, err, "expected range like 1,500, given 1200")
+
+		_, err = testutil.ExecuteCommand(cmd, "--range", "1200", "--namespace-prefix", "ns", "--namespace-range", "1,2")
+		assert.ErrorContains(t, err, "expected range like 1,500, given 1200")
+
+		_, err = testutil.ExecuteCommand(cmd, "--range", "x,y", "--namespace", "ns")
+		assert.ErrorContains(t, err, "strconv.Atoi: parsing \"x\": invalid syntax")
+
+		_, err = testutil.ExecuteCommand(cmd, "--range", "x,y", "--namespace-prefix", "ns", "--namespace-range", "1,2")
+		assert.ErrorContains(t, err, "strconv.Atoi: parsing \"x\": invalid syntax")
+
+		_, err = testutil.ExecuteCommand(cmd, "--range", "1,y", "--namespace", "ns")
+		assert.ErrorContains(t, err, "strconv.Atoi: parsing \"y\": invalid syntax")
+
+		_, err = testutil.ExecuteCommand(cmd, "--range", "1,y", "--namespace-prefix", "ns", "--namespace-range", "1,2")
+		assert.ErrorContains(t, err, "strconv.Atoi: parsing \"y\": invalid syntax")
+	})
+
+	t.Run("measure service as expected", func(t *testing.T) {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ns",
+			},
+		}
+		client := k8sfake.NewSimpleClientset(ns)
+		fakeServing := &servingv1fake.FakeServingV1{Fake: &clienttesting.Fake{}}
+		servingClient := func() (servingv1client.ServingV1Interface, error) {
+			return fakeServing, nil
+		}
+
+		fakeNetworking := &fakenetworkingv1alpha1.FakeNetworkingV1alpha1{Fake: &clienttesting.Fake{}}
+		networkingClient := func() (networkingv1alpha1.NetworkingV1alpha1Interface, error) {
+			return fakeNetworking, nil
+		}
+
+		p := &pkg.PerfParams{
+			ClientSet:           client,
+			NewServingClient:    servingClient,
+			NewNetworkingClient: networkingClient,
+		}
+
+		cmd := NewServiceMeasureCommand(p)
+
+		_, err := testutil.ExecuteCommand(cmd, "--namespace-prefix", "ns")
+		assert.ErrorContains(t, err, "no service found to measure")
+
+		_, err = testutil.ExecuteCommand(cmd, "--svc-prefix", "svc", "--namespace", "ns", "--range", "1,2", "--concurrency", "10")
+		assert.NilError(t, err)
+
+		_, err = testutil.ExecuteCommand(cmd, "--svc-prefix", "svc", "--namespace-prefix", "ns", "--concurrency", "10")
```

Comment:
Those cases cannot be put together using  the same `cmd := NewServiceMeasureCommand(p)`.
When they are putting together, the param defined in the previous command will still take effect in the current command.
E.g. `"--range", "1,2"` is defined for the second case, but when case2 and case3 are using the same cmd, when case3 does not overwrite the `--range` param, it will use the range value defined in case2. Then there will be some misunderstanding about the cases.